pkg/kubelet/pod/pod_manager.go

```diff
@@ -153,6 +158,9 @@ func (pm *basicManager) UpdatePod(pod *v1.Pod) {
 // lock.
 func (pm *basicManager) updatePodsInternal(pods ...*v1.Pod) {
 	for _, pod := range pods {
+		if pm.secretManager != nil {
```

Comment:
Yes, but doesn't the update call increment the ref count? I.e.
1. Add: ref count -> 1
2. Update: ref count -> 2
3. Delete: ref count -> 1, and is orphaned?